### DISCUSSION TITLE
Certain custom events dont fire[Issue]:

### DISCUSSION THREAD

**Discussion by tauqeer-cs** [oddbit/flutter_facebook_app_events#260] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We have made a class in which we have static method to log the fb custom event. We almost get all the events but there is some specific event that never trigger for us. For example RegisterPersonalDetails is events is never shows up on events page but we also dont get any exception too.

Example of how we call out static method 

      AnalyticService.logEvent(name: "RegisterPersonalDetails");

Class code.

```dart
import 'package:facebook_app_events/facebook_app_events.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/foundation.dart';
import 'package:gopayz/application.dart';
import 'package:logging/logging.dart';

class AnalyticService{

  final facebookAnalytics = FacebookAppEvents();

  static AnalyticService _instance = AnalyticService._();
  AnalyticService._();

  static void init() async{
    await _instance.facebookAnalytics.setAdvertiserTracking(enabled: true);
    await _instance.facebookAnalytics.setAutoLogAppEventsEnabled(true);
  }

  static void logEvent({ String ? name,  Map<String, Object?> ? parameters}) async {
    if(name.isNullOrEmpty()){
      return;
    }
    try{
      await _instance.facebookAnalytics.logEvent(name: name!,parameters: parameters);
      debugPrint("===Send Analytic: $name====");
    }catch(e){
      debugPrint("===Error Analytic: $e");
    }
  }
}
```

### Expected Behavior

The problem is commen with iOS and Android. 
Call  AnalyticService.logEvent(name: "RegisterPersonalDetails").

### Steps To Reproduce

Using   facebook_app_events: ^0.16.0
Call 

### Platform

Android

### Flutter Doctor

```shell
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-PK)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

One thing I am using fvm and this project is on 2.10.3
```

**Comment by DennisAlund** [comment]
This looks like it's implemented as it should. Most of the methods in `FacebookAppEvents` are just abstractions that call the base method `logEvent`. 


https://github.com/oddbit/flutter_facebook_app_events/blob/8bca71d848fc2ec569b74eb4ddc79c282019ab23/lib/facebook_app_events.dart#L147-L154

My guess would be that you need to either provide `valueToSum` or some parameters in the object map. I can imagine that your analytics event call doesn't do much unless you specify some data to track... just my guess.

Please add some value in the `parameters` map and let me know if the problem persists.

**Comment by tauqeer-cs** [comment]
Thanks ill add some parameters and try.

**Comment by DennisAlund** [comment]
How did it work out for you @tauqeer-cs?

**Comment by tauqeer-cs** [comment]
@DennisAlund sorry I was away for some days , yes I tried added parameter to that event which was not reaching FB but after tried it didnt work out. Its odd , it work with dev app but on prod that specific events never showed.

**Comment by DennisAlund** [comment]
Thank you for following up. Seems odd and could there be a configuration difference on your Facebook project perhaps? I'll move this issue to a discussion instead since it doesn't seem to be a technical issue or bug.

**Reply by tauqeer-cs** [reply]
Ya I have been on this for some time and ended up having a call with facebook dev team and showed them how this work and actually had to explain them how flutter packages use the native code too. Im thinking maybe test creating a native method for this and try what happens in that case. Logically everything seems to be fine otherwise.
